kratos/includes/geometrical_object.h

```diff
@@ -100,7 +102,8 @@ class GeometricalObject : public IndexedObject, public Flags, public std::intrus
     GeometricalObject(GeometricalObject const& rOther)
         : IndexedObject(rOther.Id()),
           Flags(rOther),
-          mpGeometry(rOther.mpGeometry)
+          mpGeometry(rOther.mpGeometry),
+          mReferenceCounter(0)
```

Comment:
a note to reviewers: mReferenceCounter is atomic and it cannot be assigned directly where defined (we tried)